private_sharing/models.py

```diff
@@ -403,18 +421,42 @@ def random_project_member_id():
 
         return code
 
-    def leave_project(self, remove_datafiles=False, done_by=None):
+    def deauth_webhook(self, json_p):
+        """
+        Sends a POST to an OAUTH2 project's specificed member erasure webhook URL.
+        """
+        url = self.project.oauth2datarequestproject.deauth_webhook
+
+        request_p = requests.post(url, json=json_p)
+        return request_p.status_code
+
+    def leave_project(self, remove_datafiles=False, done_by=None, erasure_requested=False):
         self.revoked = True
         self.joined = False
         self.authorized = False
+        if erasure_requested:
+            self.erasure_requested = timezone.now()
         self.save()
 
+        if self.erasure_requested == None:
```

Comment:
would it maybe be simpler to use `bool()`?

Reply:
Doh